### DISCUSSION TITLE
Client specification v2.0: Introducing individual translation archives for caching and other changes

### DISCUSSION THREAD

**Discussion by kbdharun** [tldr-pages/tldr#10709] [Announcements]
Calling all maintainers of both clients and tldr alike!

Since the [v1.5b](https://github.com/tldr-pages/tldr/releases/tag/v1.5b) release, the `master` branch was removed. The clients that haven't updated got moved to a [dedicated section in the wiki](https://github.com/tldr-pages/tldr/wiki/tldr-pages-clients#currently-broken-clients-and-integrations). If you are a user of any of these clients, we highly recommend migrating to [a supported community client or one of our official clients](https://github.com/tldr-pages/tldr/wiki/tldr-pages-clients).

### Changelog

This version of the [CLIENT SPECIFICATION](https://github.com/tldr-pages/tldr/blob/v2.0/CLIENT-SPECIFICATION.md) introduces the following changes:

- Add recommendation to support macos alias for osx (https://github.com/tldr-pages/tldr/pull/7514) (thanks @MasterOdin)
- Drop the special "all" platform from the --list flag (https://github.com/tldr-pages/tldr/pull/7561) (thanks @dbrgn)
- Drop the master branch from the assets link. (https://github.com/tldr-pages/tldr/pull/9668) (thanks @MasterOdin)
- Require support for long options (https://github.com/tldr-pages/tldr/pull/9651) (thanks @pixelcmtd)
- Add recommendation to support caching individual translation archives (https://github.com/tldr-pages/tldr/pull/10148) (thanks @kbdharun)

### Requiring support for long options

Clients will be required to add support for long options for commands along with short options for a more uniform experience.

### Introducing caching support for individual translation archives for each language

Addressing the community's requests, we have added support for individual language-specific translation archives in <https://github.com/tldr-pages/tldr/pull/10555> (thanks @adamazing). If your client displays pages based on the user's language configuration from the cache, now you can include the selected language along with English as a fallback to save space from unused languages. 

Complete translation archives will still be available for clients using it.

## tldr maintainers

Dependabot alerts and CodeQL analysis got enabled on compatible repositories org-wide.

## Community updates

- The primary chatroom address has been updated to [#tldr-pages:matrix.org](https://matrix.to/#/#tldr-pages:matrix.org). The old Gitter chatroom address was added as an alternate published address.
- Last year, we faced an incident with the CLA [bot], where we fixed the title of our project in the GitHub Gist that invalidated any previously signed CLAs of existing contributors. We are sorry for the inconvenience caused, and we suggest resigning the CLA when you contribute again.
- The domain <https://tldr.sh> has been verified in the [tldr-pages GitHub organization](https://github.com/tldr-pages). (thanks @agnivade)
- We are now on Fediverse. We have created a new account on Fosstodon (A @mastodon instance for FOSS projects). Feel free to check out <https://fosstodon.org/@tldr_pages>. (thanks @sbrl)
- The live demo web client on the official website has been updated (<https://github.com/tldr-pages/tldr-pages.github.io/pull/65>) to use <https://tldr.inbrowser.app>. (thanks @rwv)
- Introducing the new [Snap package](https://snapcraft.io/tldr) for our [Python client](https://github.com/tldr-pages/tldr-python-client). (thanks @soumyaDghosh)
- We have added [featured information](https://github.com/github/explore/pull/3805) to the `tldr` and `tldr-pages` topics on GitHub. (thanks @kbdharun)
- New maintainers and members got added with other role transitions, check out [MAINTAINERS.md](https://github.com/tldr-pages/tldr/blob/v2.0/MAINTAINERS.md) for more information.

Last year's Hacktoberfest was one of the biggest for tldr-pages, with more than 700 PRs merged from 240+ contributors🎉 . We are excited for this year's Hacktoberfest next month.

Thanks to all the contributors and maintainers for their contributions.

<hr /><em>This discussion was created from the release <a href='https://github.com/tldr-pages/tldr/releases/tag/v2.0'>Client specification v2.0: Introducing individual translation archives for caching and other changes</a>.</em>
